app/src/services/version.js

```diff
@@ -0,0 +1,168 @@
+    }
+  },
+
+  /**
+   * @function createVersionOfObjects
+   * Saves a version of each object in an array
+   * @param {object[]} data array of objects each with an `objectId` and version data
+   * @param {string} uuid of the current user
+   * @param {object} [etrx=undefined] An optional Objection Transaction object
+   * @returns {Promise<object>} The result of calling the create() method for all objects
+   * @throws The error encountered upon db transaction failure
+   */
+  createManyObjects: async (data = [], userId, etrx = undefined) => {
+    let trx;
+    try {
+      trx = etrx ? etrx : await Version.startTransaction();
+      // wait for all create version calls
+      const response = await Promise.all(data.map(async (object) => {
+        await service.create(object, userId, trx);
+      }));
+      if (!etrx) await trx.commit();
+      return response;
+    } catch (err) {
+      if (!etrx && trx) await trx.rollback();
+      throw err;
+    }
+  },
+
+  /**
+   * @function delete
+   * Delete a version record of an object
+   * @param {string} objId The object uuid
+   * @param {string} versionId The version ID
+   * @param {string} uuid of the current user
+   * @param {object} [etrx=undefined] An optional Objection Transaction object
+   * @returns {Promise<object>} The number of remaining versions in DB after the delete
+   * @throws The error encountered upon db transaction failure
+   */
+  delete: async (objId, version, currentUserId = SYSTEM_USER, etrx = undefined) => {
+    let trx;
+    try {
+      trx = etrx ? etrx : await Version.startTransaction();
+      await Version.query(trx)
+        .delete()
+        .where('objectId', objId)
+        .where('versionId', version.VersionId)
+        .throwIfNotFound();
+
+      // if other versions exist, make next most recent version current
+      const countVersions = await Version.query(trx)
+        .count('id')
+        .where({
+          objectId: objId,
+          versionId: version.VersionId
+        });
+      // TODO: get row count as integer easier way
+      const count = parseInt(countVersions[0].count);
+      if(count > 0) await service.makeLatest(objId, undefined, currentUserId, trx);
+
+      if (!etrx) await trx.commit();
+      return count;
+    } catch (err) {
+      if (!etrx && trx) await trx.rollback();
+      throw err;
+    }
+  },
+
+  /**
+   * @function makeLatest
+   * Updates most recent or provided version with isLatest = true and all other versions of object with isLatest = false
+   * @param {string} objId The object uuid
+   * @param {string} versionId The version ID
+   * @param {string} uuid of the current user
+   * @param {object} [etrx=undefined] An optional Objection Transaction object
+   * @returns {Promise<object>} The result of running the update operation
+   * @throws The error encountered upon db transaction failure
+   */
+  makeLatest: async (objId, versionId = undefined, currentUserId = SYSTEM_USER, etrx = undefined) => {
```

Comment:
This part I'm not sure we need? Tracking the latest in the database... worth a discussion

Reply:
IsLatest is a S3 construct to mark a version as 'current'. 
For example, to check if a object has been soft-deleted, you need to check if a delete-marker is the latest version.
Jeremy suggested using the createdAt field to check which one is latest instead.
I'll try removing this isLatest stuff and seeing how it goes.